File: mycat/src/main/scala/com/lorelib/mycat/service/UserService.scala
package com.lorelib.mycat.service

import com.lorelib.mycat.entity.User
import com.lorelib.mycat.mapper.UserMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.{List => JList}

/**
  * @description UserService: 
  * @author listening
  * @create 2017 07 26 上午10:06.
  */
@Service
@Transactional
class UserService @Autowired() (private val userMapper: UserMapper) extends IUserService {
  override def getAll: JList[User] = {
    userMapper.getAll
  }
}

File: learn-mybatisplus/src/main/java/com/lorelib/mybatisplus/quickstart/UserService.java
package com.lorelib.mybatisplus.quickstart;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 * @author listening
 * @description
 * @date 2017-10-09 11:35
 * @since 1.1
 */
@Service
public class UserService {
    @Autowired
    private UserMapper userMapper;

    public int insert() {
        int result = 0;
        User user = new User();
        user.setName("Tom");
        result = userMapper.insert(user);
        assert result == 1 : "插入数据失败";
        return user.getId();
    }
}
